packages/amplify_api/lib/src/graphql/graphql_request_factory.dart

```diff
@@ -0,0 +1,160 @@
+import 'package:flutter/foundation.dart';
+
+class DocumentInputs {
+  // Upper document input: ($id: ID!)
+  final String upper;
+  // Lower document input: (id: $id)
+  final String lower;
+  const DocumentInputs(this.upper, this.lower);
+}
+
+class GraphQLRequestFactory {
+  GraphQLRequestFactory._();
+
+  static final GraphQLRequestFactory _instance = GraphQLRequestFactory._();
+
+  static GraphQLRequestFactory get instance => _instance;
+
+  String _getModelType(ModelFieldTypeEnum val) {
+    switch (val) {
+      case ModelFieldTypeEnum.string:
+        return "String";
+      case ModelFieldTypeEnum.int:
+        return "Int";
+      case ModelFieldTypeEnum.model:
+        return "ID";
+      default:
+        return "Error: unsupported type!";
+    }
+  }
+
+  String _getFieldsFromModelType(ModelSchema schema) {
+    // schema has been validated & schema.fields is non-nullable
+    Map<String, ModelField> fieldsMap = schema.fields!;
+    return fieldsMap.entries
+        .map((entry) => entry.value.association == null ? entry.key : '')
+        .toList()
+        .join(' ');
+  }
+
+  ModelSchema _getAndValidateSchema(
+      ModelType modelType, GraphQLRequestOperation operation) {
+    ModelProviderInterface? provider = AmplifyAPI.instance.getModelProvider();
+
+    if (provider == null) {
+      throw ApiException('No modelProvider found',
+          recoverySuggestion:
+              'Pass in a modelProvider instance while instantiating APIPlugin');
+    }
+
+    ModelSchema schema = provider.modelSchemas.firstWhere(
+        (elem) => elem.name == modelType.modelName(),
+        orElse: () => throw ApiException(
+            'No schema found for the ModelType provided',
+            recoverySuggestion:
+                'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a valid ModelType'));
+
+    if (schema.fields == null) {
+      throw ApiException('Schema found does not have a fields property',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin');
+    }
+
+    if (operation == GraphQLRequestOperation.list &&
+        schema.pluralName == null) {
+      throw ApiException('No schema name found',
+          recoverySuggestion:
+              'Pass in a valid modelProvider instance while instantiating APIPlugin or provide a valid ModelType');
+    }
+
+    return schema;
+  }
+
+  DocumentInputs _buildDocumentInputs(
+      ModelSchema schema, GraphQLRequestOperation operation) {
+    String upperOutput = '';
+    String lowerOutput = '';
+    List<String> upperList = [];
+    List<String> lowerList = [];
+
+    // build inputs based on request operation
+    switch (operation) {
+      case GraphQLRequestOperation.get:
+        upperList.add(r"$id: ID!");
+        lowerList.add(r"id: $id");
+        lowerOutput = "(${lowerList.join(", ")})";
+        break;
+      case GraphQLRequestOperation.list:
+        break;
```

Comment:
You can remove `break` so it falls through to the default case. Don't need to fix now, just an FYI.